packages/firebase_snippets_app/lib/snippets/auth.dart

```diff
@@ -0,0 +1,622 @@
+
+    // [START manage_users_re_auth]
+    // Prompt the user to re-provide their sign-in credentials.
+    // Then, use the credentials to reauthenticate:
+    await user?.reauthenticateWithCredential(credential);
+    // [END manage_users_re_auth]
+
+    // [START manage_users_delete_user]
+    await user?.delete();
+    // [END manage_users_delete_user]
+  }
+
+  void passwordAuth_createPasswordBasedAccount() async {
+    const String emailAddress = 'user@example.com';
+    const String password = 'sdf8798324jf673qf8h!';
+
+    // [START password_auth_create_password_based_account]
+    try {
+      final credential =
+          await FirebaseAuth.instance.createUserWithEmailAndPassword(
+        email: emailAddress,
+        password: password,
+      );
+    } on FirebaseAuthException catch (e) {
+      if (e.code == 'weak-password') {
+        print('The password provided is too weak.');
+      } else if (e.code == 'email-already-in-use') {
+        print('The account already exists for that email.');
+      }
+    } catch (e) {
+      print(e);
+    }
+    // [END password_auth_create_password_based_account]
+
+    await FirebaseAuth.instance.signOut();
+
+    // [START password_auth_sign_in]
+    try {
+      final credential = await FirebaseAuth.instance.signInWithEmailAndPassword(
+        email: emailAddress,
+        password: password,
+      );
+    } on FirebaseAuthException catch (e) {
+      if (e.code == 'user-not-found') {
+        print('No user found for that email.');
+      } else if (e.code == 'wrong-password') {
+        print('Wrong password provided for that user.');
+      }
+    }
+    // [END password_auth_sign_in]
+
+    // [START password_auth_sign_out]
+    await FirebaseAuth.instance.signOut();
+    // [END password_auth_sign_out]
+  }
+
+  void emailLinkAuth_sendAuthLinkEmail() async {
+    // [START email_link_auth_action_code_settings]
+    var acs = ActionCodeSettings(
+      // URL you want to redirect back to. The domain (www.example.com) for this
+      // URL must be whitelisted in the Firebase Console.
+      url: 'https://www.example.com/finishSignUp?cartId=1234',
+      // This must be true
+      handleCodeInApp: true,
+      iOSBundleId: 'com.example.ios',
+      androidPackageName: 'com.example.android',
+      // installIfNotAvailable
+      androidInstallApp: true,
+      // minimumVersion
+      androidMinimumVersion: '12',
+    );
+    // [END email_link_auth_action_code_settings]
+
+    // [START email_link_auth_send_auth_link_email]
+    var emailAuth = 'someemail@domain.com';
+    FirebaseAuth.instance
+        .sendSignInLinkToEmail(email: emailAuth, actionCodeSettings: acs)
+        .catchError(
+            (onError) => print('Error sending email verification $onError'))
+        .then((value) => print('Successfully sent email verification'));
+    // [END email_link_auth_send_auth_link_email]
+
+    await FirebaseAuth.instance.signOut();
+  }
+
```

Comment:
I expected a snippet corresponding to [Completing sign-in in a web page](https://firebase.google.com/docs/auth/web/email-link-auth#completing_sign-in_in_a_web_page) here. Is that later? Or does that not work, even for flutter-web?

Reply:
That's on my list for later. It requires using DynamicLinks, which I need to learn about first. I have added a TODO in the code though